### ISSUE TITLE
[Arm64] Add Short Vector ABI support for HW Intrinsics

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#16021]
The types `Vector64<T>` and `Vector128<T>` are intended to map to ABI `Short Vector` fundamental types.  These should be passed in vector registers.

Current code will pass these in integer registers.

`IsShortVector` can be crudely implemented as follows:
```C#
bool Compiler::IsShortVector(CORINFO_CLASS_HANDLE hClass)
{
    return isIntrinsicType(hClass) && (getBaseTypeOfSIMDType(hClass) != TYP_UNKNOWN);
}
```

`Compiler::getPrimitiveTypeForStruct()` can be trivially updated to return `TYP_SIMD8` or `TYP_SIMD16` when `IsShortVector() == true`

I made an initial naive attempt to implement and failed, but I really need a design discussion.

@CarolEidt @tannergooding @dotnet/arm64-contrib @dotnet/jit-contrib 

category:reach
theme:intrinsics
skill-level:expert
cost:medium

**Comment by sdmaclea** [comment]
#15943

**Comment by sdmaclea** [comment]
I an issue with Vector64<T> return again.

Rather than hacking it.  I am thinking about fixing this.

**Comment by BruceForstall** [comment]
@sdmaclea Is this fixed already? https://github.com/dotnet/coreclr/pull/16310 is merged.

If not, @CarolEidt, maybe this is related to your ABI work https://github.com/dotnet/coreclr/issues/14371

**Comment by sdmaclea** [comment]
@BruceForstall #16310 was abandoned (not merged).  It never passed all tests and it diverged with the master branch enough that I didn't have time to fix it.

I think @CarolEidt is working on it.  Probably the #14371 is part of it.

**Comment by CarolEidt** [comment]
I am working on #14371, and also #16022, though progress is slow.

**Comment by CarolEidt** [comment]
Fixed with #23675
